src/configurationProvider.ts

```diff
@@ -153,14 +151,26 @@ export class CSharpConfigurationProvider implements vscode.DebugConfigurationPro
 	 */
     resolveDebugConfiguration(folder: vscode.WorkspaceFolder | undefined, config: vscode.DebugConfiguration, token?: vscode.CancellationToken): vscode.ProviderResult<vscode.DebugConfiguration> {
 
-        if (!config.type)
-        {
+        if (!config.type) {
             // If the config doesn't look functional force VSCode to open a configuration file https://github.com/Microsoft/vscode/issues/54213
             return null;
         }
 
-        if (config.request === "launch")
-        {
+        if (config.request === "attach" && config.processCommand) {
+            const processCommand = config.processCommand.replace(/\${workspaceFolder}/g, folder.uri.fsPath);
+            delete config.processCommand;
+            return this.attachItemsProvider.getAttachItems()
+                .then(attachItems => {
+                    const attachItem = attachItems.find(ai => ai.detail === processCommand);
+                    if (attachItem == null) {
+                        throw new Error("Can't find process");
```

Comment:
>"Can't find process" [](start = 40, length = 20)

We might want to mention what we are looking for.

Reply:
Now throws `Couldn't find process with command ${processCommand}`